Parse/src/main/java/com/parse/ParseQuery.java

```diff
@@ -1699,6 +1701,31 @@ public void getInBackground(final String objectId, final GetCallback<T> callback
    * {@code ParseQuery}.
    * <p/>
    * This only works on keys whose values are {@link ParseObject}s or lists of {@link ParseObject}s.
+   * Add a constraint to the query that requires a particular key's value to contain each one of
+   * the provided list of values entirely or just starting with given strings.
+   *
+   * @param key
+   *          The key to check. This key's value must be an array.
+   * @param values
+   *          The values that will match.
+   * @return this, so you can chain this call.
+   */
+  public ParseQuery<T> whereContainsAllStartsWith(String key, Collection<String> values) {
```

Comment:
It took me a while to understand what this does. Personally I think the cleanest and simplest wording is to keep `whereStartsWith`:

- `whereStartsWith(key, String prefix):` check that value in `key` (must be `String`) starts with prefix
- `whereStartsWith(key, Collection<String> prefixes)`: check that values in `key` (must be `Array`) start with prefixes in the given collection (implicit: all of the constraints must be satisfied)